tests/conftest.py

```diff
@@ -1,28 +1,99 @@
+import os
+import urllib
 from collections.abc import Generator
+from multiprocessing import Process
+from time import sleep
+from typing import Any
+from urllib.error import URLError
 
 import pytest
+import uvicorn
+from _pytest.fixtures import SubRequest
 from fastapi.testclient import TestClient
-from sqlmodel import Session, SQLModel, create_engine
-from sqlmodel.pool import StaticPool
+from playwright.sync_api import Page, Playwright, sync_playwright
+from sqlmodel import Session
+from tad.core.config import settings
+from tad.core.db import get_engine
 from tad.main import app
 
-# needed to make sure create_all knows about the models
-from tad.models import *  # noqa: F403
+from tests.database_test_utils import DatabaseTestUtils
+
+
+class TestSettings:
+    HTTP_SERVER_SCHEME: str = "http://"
+    HTTP_SERVER_HOST: str = "127.0.0.1"
+    HTTP_SERVER_PORT: int = 8000
+    TIME_OUT: int = 10
+
+
+def run_server() -> None:
+    uvicorn.run(app, host=TestSettings.HTTP_SERVER_HOST, port=TestSettings.HTTP_SERVER_PORT)
+
+
+def wait_for_server_ready(server: Generator[Any, Any, Any]) -> None:
+    for _ in range(TestSettings.TIME_OUT):
+        try:
+            # we use urllib instead of playwright, because we only want a simple request
+            #  not a full page with all assets
+            assert urllib.request.urlopen(server).getcode() == 200  # type: ignore # noqa
+            break
+        # todo (robbert) find out what exception to catch
+        except URLError:  # server was not ready
+            sleep(1)
+
+
+@pytest.fixture(scope="module")
+def server() -> Generator[Any, Any, Any]:
+    # todo (robbert) use a better way to get the test database in the app configuration
+    os.environ["APP_DATABASE_FILE"] = "database.sqlite3.test"
+    process = Process(target=run_server)
```

Comment:
i think it would be better to pass a custom environment to the Process function.

Reply:
Would be nice. I don't see how to pass an env to a Process in Python. The current solution is a bit of a work-around too, so will keep this for now.